src/components/PDFView/index.native.js

```diff
@@ -33,34 +35,42 @@ class PDFView extends Component {
             shouldAttemptPdfLoad: true,
             shouldShowLoadingIndicator: true,
             isPasswordInvalid: false,
+            failedToLoadPDF: false,
             password: '',
         };
         this.initiatePasswordChallenge = this.initiatePasswordChallenge.bind(this);
         this.attemptPdfLoadWithPassword = this.attemptPdfLoadWithPassword.bind(this);
         this.finishPdfLoad = this.finishPdfLoad.bind(this);
+        this.handleFailureToLoadPDF = this.handleFailureToLoadPDF.bind(this);
     }
 
     componentDidUpdate() {
         this.props.onToggleKeyboard(this.props.isShown);
     }
 
+    handleFailureToLoadPDF(error) {
+        if (error.message.match(/password/i)) {
+            this.initiatePasswordChallenge();
```

Comment:
Curious, what is this check doing?

Reply:
This was from the previous code to handle password protected PDFs. I didn't test it, but I'm guessing that if the PDF is password protected the error message will contain the word password.